### ISSUE TITLE
Double trigger in `throttle` and `debounce` with `sample`-chain 

### ISSUE THREAD

**Issue by igorkamyshev** [effector/patronum#250]
I've added tests with reproduce in #249

**Comment by stavrogyn** [comment]
It seems like the problem is not with the `throttle` and `debounce` methods themselves but with the test design that you provide in the linked pull request.

If I got correctly what problem you are trying to point out, there should be a few corrections in the pull request:

1. If we want to estimate `throttle` or `debounce` influence, we have to check the appropriate `Event` that these methods return:

```ts
const listener = jest.fn();

const debounced = debounce({ source: someEvent, timeout: 100 });

debounced.watch(listener);
```

2. We have to not use `Promise.all` with `allSettled ` checking these methods because these methods work asynchronously. Each `allSettled ` would wait for finishing all the triggered effects changing `scope` in its own way. In these cases, we should use `Promise.race` or call `allSettled` sequentially without `await` and adding `await` only for the last `allSettled`:

```ts
await Promise.race([allSettled(eventToCheckDebounced, { scope }), allSettled(eventToCheckThrottled, { scope }), allSettled(anotherEventToCheckDebounced, { scope })]);

// or

allSettled(eventToCheckDebounced, { scope })

allSettled(eventToCheckThrottled, { scope })

await allSettled(anotherEventToCheckDebounced, { scope })
```

**Comment by AlexandrHoroshih** [comment]
Fixed in `patronum@1.13.0`
